api/users/views.py

```diff
@@ -603,6 +611,63 @@ def create(self, request, *args, **kwargs):
         return Response(status=status.HTTP_204_NO_CONTENT)
 
 
+class UserChangePassword(JSONAPIBaseView, generics.CreateAPIView, UserMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        CurrentUser,
+    )
+
+    required_read_scopes = [CoreScopes.NULL]
+    required_write_scopes = [CoreScopes.USER_SETTINGS_WRITE]
+
+    view_category = 'users'
+    view_name = 'user_password'
+
+    serializer_class = UserChangePasswordSerializer
+    throttle_classes = (ChangePasswordThrottle, )
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = self.get_user()
+        existing_password = request.data['existing_password']
+        new_password = request.data['new_password']
+
+        if not user.check_password(existing_password):
```

Comment:
I don't think this is necessary (it's not done in the v1 view). It looks like `change_password` already increments `old_password_invalid_attempts`.